components/raftstore/src/store/worker/pd.rs

```diff
@@ -1413,10 +1444,73 @@ fn send_destroy_peer_message<EK, ER>(
     }
 }
 
+fn collect_report_read_peer_stats(
+    capacity: usize,
+    mut report_read_stats: HashMap<u64, pdpb::PeerStat>,
+    mut stats: pdpb::StoreStats,
+) -> pdpb::StoreStats {
+    if report_read_stats.len() < capacity * 3 {
+        for (_, read_stat) in report_read_stats {
+            let mut peer_stat = pdpb::PeerStat::default();
+            peer_stat.set_region_id(read_stat.region_id);
+            peer_stat.set_read_bytes(read_stat.read_bytes);
+            peer_stat.set_read_keys(read_stat.read_keys);
+            peer_stat.set_query_stats(read_stat.get_query_stats().clone());
+            stats.peer_stats.push(peer_stat);
+        }
+        return stats;
+    }
+    let mut peer_stats = HashMap::default();
+    let mut keys_topn_report = TopN::new(capacity);
+    let mut bytes_topn_report = TopN::new(capacity);
+    let mut stats_topn_report = TopN::new(capacity);
+    for read_stat in report_read_stats.values() {
+        let mut cmp_stat = PeerCmpReadStat::default();
+        cmp_stat.region_id = read_stat.region_id;
+        let mut key_cmp_stat = cmp_stat.clone();
+        key_cmp_stat.report_stat = read_stat.read_keys;
+        keys_topn_report.push(key_cmp_stat);
+        let mut byte_cmp_stat = cmp_stat.clone();
+        byte_cmp_stat.report_stat = read_stat.read_bytes;
+        bytes_topn_report.push(byte_cmp_stat);
+        let mut query_cmp_stat = cmp_stat.clone();
+        query_cmp_stat.report_stat = get_read_query_num(read_stat.get_query_stats());
+        stats_topn_report.push(query_cmp_stat);
+    }
+
+    for x in keys_topn_report {
+        if let Some(report_stat) = report_read_stats.remove(&x.region_id) {
+            peer_stats.insert(x.region_id, report_stat);
+        }
+    }
+
+    for x in bytes_topn_report {
+        if let Some(report_stat) = report_read_stats.remove(&x.region_id) {
+            peer_stats.insert(x.region_id, report_stat);
+        }
+    }
+
+    for x in stats_topn_report {
+        if let Some(report_stat) = report_read_stats.remove(&x.region_id) {
+            peer_stats.insert(x.region_id, report_stat);
+        }
+    }
+
+    for (_, report_stat) in peer_stats {
+        stats.peer_stats.push(report_stat);
+    }
```

Comment:
```suggestion
    for x in stats_topn_report {
        if let Some(report_stat) = report_read_stats.remove(&x.region_id) {
            stats.peer_stats.push(report_stat);
        }
    }
```

Reply:
I think we should keep `peer_stats` in order to remove duplicated peer stat in topN.

Reply:
With `report_read_stats.remove` there is not duplicated peer stat

Reply:
updated.